polyfill/intersectionobserver-polyfill.js

```diff
@@ -0,0 +1,250 @@
+/*
+Copyright 2016 Google Inc. All Rights Reserved.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+   http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+// TODO: rootMargin are completely ignored for now
+(function(scope) {
+  if('IntersectionObserver' in window) {
+    return;
+  }
```

Comment:
I agree with this, though I wouldn't like to overlap with a native implementation neither. In that case it would be necessary to see what is the partial part of the spec that's out and detect that and fix it like in `classList` API which Opera and IE10 doesn't support multiple class names or second argument in `toggle` method.